### ISSUE TITLE
Broken @update directive

### ISSUE THREAD

**Issue by Rockson** [nuwave/lighthouse#1372]
**Describe the bug**

The last version v4.13.0 breaks  the @update directive when not using "id" as primary key in the model.
While the version v4.12.4 works fine
The documentation states: 

> Lighthouse uses the argument id to fetch the model by its primary key. **This will work even if your model has a differently named primary key**, so you can keep your schema simple and independent of your database structure.

So having a primary key other than "id" should be supported.

As of now the directive is setting the column ID even though there is no ID Column in the database

**Steps to reproduce**

> type MyModel  {
    id: ID! @rename(attribute: "my_different_id")
    content: String
}

> myedit(id: ID!, content: String!): MyModel @update

> class MyModel {
        protected $table = "my_table";
        protected $primaryKey = "my_different_id";
        protected $fillable = ["content"];
}

**Output/Logs**
The generated SQL is:

 (SQL: update "my_table" set "CONTENT" = CONTENT, "ID" = 1022 where "MY_DIFFERENT_ID" = 1022)

So basically, the new version is adding ID = ? to the generated query, even though the column is not the primary key

**Lighthouse Version**

v4.13.0

**Comment by webard** [comment]
Same problem in my application, but I think it is Eloquent's forceFill method problem, not lighthouse.

**Comment by Rockson** [comment]
I think Eloquent's forceFill behaviour is intended, that's why it's called **force**. If you pass it a property that doesn't exist on the table the model has no way to know it so it creates the wrong sql. The mistake is on the lighthouse side

**Comment by webard** [comment]
You may be right @Rockson  but when you change primaryKey name, forceFill method should respect your setting. On the other side, you may have `$primaryKey = 'primary_id'` column and `id` column for something else. It is messy, but possible. One of solutions is "in the middle" method - forceFill with respecting $primaryKey.

**Comment by Rockson** [comment]
> You may be right @Rockson but when you change primaryKey name, forceFill method should respect your setting. On the other side, you may have `$primaryKey = 'primary_id'` column and `id` column for something else. It is messy, but possible. One of solutions is "in the middle" method - forceFill with respecting $primaryKey.

Yeah it should never happen that in your db the actual primary key is called "primary_id" and then in the same table you have another column named "id", but I don't think Eloquent should assert this rule in his code, it may happen on some legacy code, who knows.

**Comment by spawnia** [comment]
I see how switching to using `forceFill()` caused your code to break. In hindsight, i wish i considered that case and been more cautious.

Now that the change already landed in `v4.13`, reverting it might cause breakage in the other direction. Ultimately, i believe that the whole idea of mass-assignment protection is not that great so i don't necessarily want to build features based on the assumption that it is used at all.

You can fix your code by explicitly applying `@rename` to your argument:

```diff
myedit(
- id: ID!
+ id: ID! @rename(attribute: "actual_primary_key_column")
  content: String!
): MyModel @update
```

**Comment by spawnia** [comment]
I just thought of a way to fix this, see https://github.com/nuwave/lighthouse/pull/1377

**Comment by Rockson** [comment]
Ok but as I understand, now I have some kind of repetition no?

`
Type MyModel{
Id:  ID @rename(name: "primary_id")
}

Edit(id: ID @rename(name: "primary_id)) @update
`

Can't we get rid of it?

Anyway thank for the reply :)

**Comment by spawnia** [comment]
Releases in https://github.com/nuwave/lighthouse/releases/tag/v4.13.1
